### DISCUSSION TITLE
use circleci sdk for generate circleci content

### DISCUSSION THREAD

**Discussion by karlderkaefer** [projen/projen#2346] [General]
I just found there is a [SDK for circleci](https://github.com/CircleCI-Public/circleci-config-sdk-ts). This library is fully generating the whole pipeline [as string](https://github.com/CircleCI-Public/circleci-config-sdk-ts/blob/79e0b77c03487baac74899d0cfd75f710954766d/src/lib/Config/index.ts#L235-L237), probably it can replace the existing code and give better user experience since they have a better typed model. Also then we do not need to care about API updates

**Comment by mrgrain** [comment]
Yes probably. Although the SDK might not be jsii compatible (*). Long term I would like to see these kind of providers live in separate packages, with a solid integration pattern so that users can (more or less) plug-n-play different workflow providers.


(*) We can still use it, but it would require a jsii compatible wrapper around it.
